src/main/webapp/search-script.js

```diff
@@ -0,0 +1,122 @@
+function collectUnitsFromSearch() {
+  const inputElems = document.getElementsByName("search-units");
+  const addedUnits = []
+  for (let elem of inputElems) {       
+    if (elem.checked){
+      addedUnits.push(elem.value);
+    }
+  }
+  return addedUnits;
+}
+
+function collectUnitsFromInput() {
+  const inputElems = document.getElementsByName("num-units");
+  const addedUnits = []
+  for (let elem of inputElems) {       
+    if (elem.checked){
+      addedUnits.push(elem.value);
+    }
+  }
+  return addedUnits;
+}
+
+function showCourses() {
+  const courseResults = document.getElementById("search-results");
+  const courseName = document.getElementById("search-course").value;
+  const profName = document.getElementById("search-prof").value;
+  const termName = document.getElementById("search-term").value;
+  const units = collectUnitsFromSearch();
+  const school = getUserSchool();
+  courseResults.innerHTML = "";
+  const url = new URL("/search", window.location.origin);
+  url.searchParams.set("courseName", courseName);
+  url.searchParams.set("profName", profName);
+  url.searchParams.set("units", units);
+  url.searchParams.set("term", termName);
+  url.searchParams.set("schoolName", school)
+  fetch(url)
+  .then(response => response.json())
+  .then((searchResults) => {
+    if (searchResults.hasOwnProperty("message")) { // TODO: Make a div for this message so specific styling is easier.
```

Comment:
Since you're using async await later, might be good to convert this to it as well. Typically good to keep the file as consistent as possible with using async/await or avoiding it.